compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -3221,7 +3221,7 @@ impl<'ast, 'ra, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
                                 if i != 0 {
                                     binders += ", ";
                                 }
-                                binders += x.as_str();
+                                binders += &x.to_string();
```

Comment:
```suggestion
                                let _ = write!(binders, "{x}");
```
I think this will allow writing the formatted `x` _into_ `binders`, instead of creating a whole separate `String`, appending it to `binders`, and then throwing it away